support-workers/src/main/scala/com/gu/zuora/ProductSubscriptionBuilders.scala

```diff
@@ -5,37 +5,26 @@ import java.util.UUID
 import com.gu.config.Configuration
 import com.gu.i18n.Country
 import com.gu.support.catalog
-import com.gu.support.catalog.{Product, ProductRatePlan, ProductRatePlanId}
-import com.gu.support.config.TouchPointEnvironments.UAT
+import com.gu.support.catalog.{ProductRatePlan, ProductRatePlanId}
 import com.gu.support.config.{TouchPointEnvironments, ZuoraConfig}
 import com.gu.support.promotions.{PromoCode, PromotionService}
-import com.gu.support.workers.exceptions.{BadRequestException, CatalogDataNotFoundException}
+import com.gu.support.workers.ProductTypeExtensions._
 import com.gu.support.workers._
+import com.gu.support.workers.exceptions.{BadRequestException, CatalogDataNotFoundException}
 import com.gu.support.zuora.api._
 import org.joda.time.{DateTimeZone, LocalDate}
 
 import scala.util.{Failure, Success, Try}
 
 object ProductSubscriptionBuilders {
 
-  def getProductRatePlanId[PT <: ProductType, P <: Product](product: P, productType: PT, isTestUser: Boolean): ProductRatePlanId = {
-    val touchpointEnvironment = if (isTestUser) UAT else TouchPointEnvironments.fromStage(Configuration.stage)
-
-    val ratePlans: Seq[ProductRatePlan[Product]] = product.ratePlans.getOrElse(touchpointEnvironment, Nil)
-
-    val maybeProductRatePlanId: Option[ProductRatePlanId] = productType match {
-      case c: Contribution => ratePlans.find(rp => rp.billingPeriod == c.billingPeriod).map(_.id)
-      case dp: DigitalPack => ratePlans.find(rp => rp.billingPeriod == dp.billingPeriod).map(_.id)
-      case p: Paper => ratePlans.find(rp => rp.fulfilmentOptions == p.fulfilmentOptions && rp.productOptions == p.productOptions).map(_.id)
-      case gw: GuardianWeekly => ratePlans.find(rp => rp.billingPeriod == gw.billingPeriod && rp.fulfilmentOptions == gw.fulfilmentOptions).map(_.id)
-      case _ => None
-    }
+  private def getTouchPointEnvironment(isTestUser: Boolean) = TouchPointEnvironments.fromStage(Configuration.stage, isTestUser)
 
-    Try(maybeProductRatePlanId.get) match {
+  def validateRatePlan(maybeProductRatePlan: Option[ProductRatePlan[catalog.Product]], productDescription: String): ProductRatePlanId =
+    Try(maybeProductRatePlan.map(_.id).get) match {
```

Comment:
Why `.get` instead of a direct match on the `Option`? We're introducing a `Try` largely for the sake of it.